### ISSUE TITLE
[RFE] Create OpenAPI Spec

### ISSUE THREAD

**Issue by StykMartin** [beaker-project/beaker#193]
**Is your feature request related to a problem? Please describe.**
The Beaker Server and Beaker Labcontroller API are not properly documented.

**Describe the solution you'd like**
Create [OpenAPI Spec](https://swagger.io/specification/) for Beaker Server and Beaker Labcontroller. 
This will be the source of truth for implementation.

**Describe alternatives you've considered**
Avoid writing documentation, but I felt like people might not like it.

**Comment by StykMartin** [comment]
Based on the discussion we had, I think you might want to take a look at this @p3ck

**Comment by mdujava** [comment]
Some generated documentation can be accessed on `http://lab-controller-fqdn:8000/server`.
It would be ideal to get openapi spec generated from xmlrpc/turbogears directly, but few minutes searching on web did not reveal any out-of-the-box tool for it.

**Comment by StykMartin** [comment]
The goal is to get rid of XMLRPC and Turbogears, so tooling is not necessary in this case.

**Comment by StykMartin** [comment]
But good point @mdujava, I updated the description to make that clear.

**Comment by mdujava** [comment]
Yeah, I would say when creating new REST api that would preserve functionality, would be easier and less problems would be introduced when existing API would be somehow defined (eg. list of all methods/endpoints).
